src/diffusers/models/transformers/transformer_2d.py

```diff
@@ -153,94 +149,178 @@ def __init__(
                 f" {patch_size}. Make sure that `in_channels`, `num_vector_embeds` or `num_patches` is not None."
             )
 
-        # 2. Define input layers
-        if self.is_input_continuous:
-            self.in_channels = in_channels
+        self.in_channels = in_channels
```

Comment:
it seems like for continuous input, out_channnels should always be same as in_channels, not sure if we should add a check for it
```suggestion
        self.in_channels = in_channels
        self.out_channels = in_channels if out_channels is None else out_channels
        self.gradient_checkpointing = False
```